src/Cassandra/Mapping/TypeConversion/TypeConverter.cs

```diff
@@ -61,6 +61,42 @@ protected TypeConverter()
             return converter(value);
         }
 
+        internal object ConvertObjectValue(Type valueType, Type dbType, object value)
+        {
+            var converter = GetToDbConverter(valueType, dbType);
+            if (converter == null)
+            {
+                try
+                {
+                    return Convert.ChangeType(value, dbType);
```

Comment:
We could throw an exception in the `TypeConverter`. 
When converting the db / udt field value, we would check if the conversion is necessary using `IsAssignableFrom()` and only perform it in that case.
If something goes wrong, the exception will be thrown and the user will know the cause of the error.

Is that what you were thinking about?